opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -55,3 +58,45 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+
+
+class ResourceDetector:
+    def __init__(self, crash_on_error=False):
+        self.crash_on_error = crash_on_error
+
+    # pylint: disable=no-self-use
+    def detect(self) -> "Resource":
+        return _EMPTY_RESOURCE
+
+
+class OTELResourceDetector:
```

Comment:
```suggestion
class OTResourceDetector:
```

Reply:
@james-bebbington Can you comment?

Reply:
As below, I believe we can resolve this for now, it's easy to go back and change it if necessary.